src/receive_imf.rs

```diff
@@ -1049,6 +1049,37 @@ async fn add_parts(
         .cloned()
         .unwrap_or_default();
 
+    let supersedes_msg_id = match mime_parser.get_header(HeaderDef::Supersedes) {
+        Some(supersedes) => {
+            if let Some(msg_id) = rfc724_mid_exists(context, supersedes).await? {
+                if let Some(orig_from_id) = context
+                    .sql
+                    .query_row_optional("SELECT from_id FROM msgs WHERE id=?", (msg_id,), |row| {
+                        let from_id: ContactId = row.get(0)?;
+
+                        Ok(from_id)
+                    })
+                    .await?
+                {
+                    if from_id == orig_from_id {
+                        Some(msg_id)
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        }
+        None => None,
+    };
+    if supersedes_msg_id.is_some() {
+        replace_msg_id = supersedes_msg_id;
+        info!(context, "Superseding {supersedes_msg_id:?}");
+    }
```

Comment:
This is ignoring the `Supersedes` header if we don't know the superseded message.

As we talked about in reallife, if we don't know the superseded message, the `rfc_724_mid` of the new Message needs to be set to the `Supersede` header's value so that both chat partner reference it as the same `rfc_724_mid`, it can be replaced again, and so on.